applications/nrf_desktop/src/modules/ble_bond.c

```diff
@@ -647,7 +675,7 @@ static bool storage_data_is_valid(void)
 		return false;
 	}
 
-	if (cur_peer_id >= DONGLE_PEER_ID) {
+	if (get_app_id() >= DONGLE_PEER_ID) {
```

Comment:
Here you actually check if `cur_peer_id` has valid value. I think you should refer to `cur_peer_id` directly.